controllers/clusterresources/helpers.go

```diff
@@ -67,6 +70,31 @@ func areEncryptionKeyStatusesEqual(a, b *v1beta1.AWSEncryptionKeyStatus) bool {
 	return true
 }
 
+func CheckIfUserExists(username, clusterID, app string, api instaclustr.API) (bool, error) {
+	users, err := FetchUsers(clusterID, app, api)
+	if err != nil {
+		return false, err
+	}
+
+	return slices.Contains(users, username), nil
+}
+
+func FetchUsers(clusterID, app string, api instaclustr.API) ([]string, error) {
+	users := make([]string, 0)
+
+	b, err := api.FetchUsers(clusterID, app)
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.Unmarshal(b, &users)
+	if err != nil {
+		return nil, err
+	}
+
+	return users, nil
+}
```

Comment:
This function seems redundant. All this code may be moved to `api.FetchUsers(clusterID, app)`

Reply:
fixed